plugins/providers/virtualbox/driver/version_5_0.rb

```diff
@@ -587,6 +587,10 @@ def read_guest_additions_version
 
         def read_guest_ip(adapter_number)
           ip = read_guest_property("/VirtualBox/GuestInfo/Net/#{adapter_number}/V4/IP")
+          if ip.end_with?(".1")
+            @logger.warn("VBoxManage guest property returned: #{ip}. Result resembles IP of DHCP server. Retrying IP lookup.")
+            ip = read_guest_property("/VirtualBox/GuestInfo/Net/#{adapter_number}/V4/IP")
```

Comment:
Instead of reading the value again, the log warning should note the value received and that it is being ignored. Then the `ip` value should be set to `nil`(which will result in an exception being raised when the value is validated).